src/runtime/composables/useSlideover.ts

```diff
@@ -0,0 +1,61 @@
+import { ref, inject } from 'vue'
+import { createSharedComposable } from '@vueuse/core'
+import type { ShallowRef, Component, InjectionKey } from 'vue'
+import type { ComponentProps } from '../types/component'
+import type { Slideover, SlideoverState } from '../types/slideover'
+
+export const slidOverInjectionKey: InjectionKey<ShallowRef<SlideoverState>> =
+    Symbol('nuxt-ui.slideover')
+
+function _useSlideover () {
+    const slideoverState = inject(slidOverInjectionKey)
+    const isOpen = ref(false)
+
+    function open<T extends Component> (component: T, props?: Slideover & ComponentProps<T>) {
+        if (!slideoverState) {
+            throw new Error('useSlideover() is called without provider')
+        }
+
+        slideoverState.value = {
+            component,
+            props: props ?? {}
+        }
+
+        isOpen.value = true
+    }
+
+    function close () {
+        if (!slideoverState) return
+
+        isOpen.value = false
+        setTimeout(() => {
+            slideoverState.value = {
+                component: 'div',
+                props: {}
+            }
+        }, 300)
```

Comment:
This is a dirty hack to prevent the slideover from losing its exist animation. I'm open to alternatives.

Reply:
Why is this not necessary in `useModal`?

Reply:
Did you go on the `Edge` package?
![CleanShot 2024-03-04 at 16 55 36@2x](https://github.com/nuxt/ui/assets/739984/673a4373-89c8-4c3f-a831-778b35de89b4)

Reply:
That was it. thx :)